### ISSUE TITLE
Split Sass variables into more granular files

### ISSUE THREAD

**Issue by sawyerh** [uswds/uswds#2007]
I'd like to propose an idea that would make it easier for projects extending the WDS to import just the Sass variables relevant to them.

## Current pain point

Right now all Sass variables live in a single file: [`core/_variables.scss`](https://github.com/18F/web-design-standards/blob/develop/src/stylesheets/core/_variables.scss). This includes variables for typography, color, layout, assets, and magic numbers. This is a pain point for our project in particular, because:

- We're unable to use `rem` units, so we want to define our own type scale variables, and
- we have our own (flexbox) grid framework, so we want to define our own layout variables
- so, ideally there'd be a way to just import the color variables without everything else.

## Proposed solution

I'd like to propose a change to the directory structure, introducing a new `settings` directory which holds more granular variables files. I think the following structure could work:

```
└── stylesheets
    ├── components
    ├── core
    ├── elements        
    ├── settings
    │   ├── _variables.assets.scss        Asset pipeline and font/image paths
    │   ├── _variables.color.scss         Colors
    │   ├── _variables.layout.scss        Grid, breakpoints, spacing
    │   ├── _variables.misc.scss          Magic numbers, etc
    │   └── _variables.typography.scss    Type scale and font stack  
```

WDS could continue to have the `core/_variables.scss` for backwards-compatibility reasons, and it would `@import` the individual files mentioned above.

This structure would allow other Sass projects to incrementally implement the WDS. For example:

```sass
@import 'uswds/settings/variables.color';

body {
  color: $color-base;
  font-family: 'Proxima Nova';
}

// ...more custom styles

```

This may also help with #960, allowing you to link people to just the `_variables.color.scss` file.

----

I'd be happy to make this PR, but want to get your thoughts before putting in the work 😄

**Comment by AndreaSigz** [comment]
Thanks @sawyerh! @donjo or @toolness any thoughts?

**Comment by sawyerh** [comment]
👋 Heya, just wanted to bump this thread. This is still a pain point we run into, and I'd be happy to make this PR if it's something y'all would be interested in.

**Comment by maya** [comment]
Hi @sawyerh, yes we definitely would like to look into this. The current sprint is focused on small to medium sized bugs. Since variables are central to how the system works, this would likely have to be tackled more comprehensively (with some research and user testing) to ensure we still have a strong, easy-to-use developer experience. 

Currently, many developers rely on the `variables.scss` file as a single place to find all the variables, so would like to make sure any changes we make still allows for a straightforward developer experience. If you have thoughts around this point, feel free to share.

**Comment by thisisdano** [comment]
I'm open to this — let's look at potential solutions as we synchronize and square existing USWDS variables with the new 2.0 variables.

I suppose we could have a `_settings-all.scss` that collects all the settings files together and also break out individual settings files more granularly — just as we're going with `uswds-components`, `uswds-fonts`, `uswds-layout-grid`, and `uswds-utilities` packages that are collected into `uswds` with 

```scss
// -------------------------------------
// import specific USWDS modules...

// @import 'uswds-fonts';
// @import 'uswds-layout-grid';
// @import 'uswds-components';
// @import 'uswds-utilities';

// -------------------------------------
// ...or import all USWDS modules

@import 'uswds';
```

**Comment by maya** [comment]
Resolved in #2687. 

@sawyerh feel free to take a look.

**Comment by sawyerh** [comment]
![image](https://user-images.githubusercontent.com/371943/44812048-3bbc9100-aba4-11e8-8af4-e38359222d7f.png)
